python/paddle/fluid/dygraph/layers.py

```diff
@@ -865,30 +865,30 @@ def _build_once(self, *args, **kwargs):
         pass
 
     def __call__(self, *inputs, **kwargs):
-        for forward_pre_hook in self._forward_pre_hooks.values():
-            hook_result = forward_pre_hook(self, inputs)
-            if hook_result is not None:
-                if not isinstance(hook_result, tuple):
-                    hook_result = (hook_result, )
-                inputs = hook_result
-
-        if not self._built:
-            with program_desc_tracing_guard(False):
-                self._build_once(*inputs, **kwargs)
-                if parallel_helper._is_data_parallel_mode():
-                    parallel_helper._broadcast_parameters(
-                        self._parameters.values())
-            self._built = True
-
         with param_guard(self._parameters), param_guard(self._buffers):
+            for forward_pre_hook in self._forward_pre_hooks.values():
```

Comment:
A question: Is this within expectations？
It seems like that the **parameters** in hooks can be  in `param_guard`, then the static graph can cover the parameters. But the operators in hooks may not be convered in static graph if this Layer is the outermost layer which is decorated by `paddle.jit.to_static`.

Reply:
It is within expectation but you are also right. We haven't supported paddle.jit.to_static in hook now and we don't think we have enough time for 2.0RC, so this PR just moved `param_guard` without handling dy2stat for hooks. We will do that in the future.